### PR TITLE
Adding optimizations for tttt in the 2lss SR

### PR DESCRIPTION
**Pull request by Dovahkinsl**
Adding the optimal sectioning of the 2lss SR based on loose and medium bjets

### CODE REVIEW

**Comment by kmohrman** [comment]
Thanks for this PR @Dovahkinsl. I'm wondering if you could report the percent improvement you see when running with a differential variable? 

I think the CI reference files will also need to be updated.

File: analysis/topEFT/datacard_maker.py [hunk 1]
```diff
@@ -221,6 +227,7 @@ def export1d(h, name, cat, fout):
         def export2d(h):
             return h.to_hist().to_numpy()
         if isinstance(channel, str) and channel not in self.channels:
+           print(channel)
```
[the remark this answers is not in the document]
Comment (Dovahkinsl) [reply analysis/topEFT/datacard_maker.py:230]: Sure, this is a leftover of some debugging

**Comment by Dovahkinsl** [comment]
Applying the bugfix in the datacardmaker actually improved sensitivity.
We are looking at around 12% improvement for the 4 heavy quark operators when all other WC are floating and around 18% improvement for the 4 heavy quark operators when all other WC are frozen at 0.0 at 1 sigma. This is compared to what is currently on the master branch. Compared to what we had before any optimization to the four heavy quark operators we are looking at 25% (other WC floating) and 33% (other WC frozen) respectively. I will post the 2 sigma improvements soon as well.

**Comment by kmohrman** [comment]
Thanks very much for the numbers @Dovahkinsl, that is great news. Sorry to be so specific, but I'm wondering if you could also provide:
* The percent improvement (compared to the current master branch) at 2 sigma, (as this is what we usually quote)? 
* The details of the measurement (i.e. exactly which distributions you were using in which categories)
*  The actual 2 sigma limits (either before or after the improvement) so that I can compare against what I am currently getting, just to make sure there are not unexpected differences? 

Sorry for all of the requests, I'm just trying to make sure we know all of the expected changes before we merge, and that there is enough detail to know that the results would be is reproducible by someone else.

**Comment by kmohrman** [comment]
Something like this would be ideal, if it's not too much of a hassle:
```
2sig_lims_master_lj0pt = {
    "wc1" : [low_lim, hi_lim],
    "wc2" : [low_lim, hi_lim],
    ...
}
2sig_lims_thisbranch_lj0pt = {
    "wc1" : [low_lim, hi_lim],
    "wc2" : [low_lim, hi_lim],
    ...
}
```

File: analysis/topEFT/datacard_maker.py [hunk 2]
```diff
@@ -144,7 +144,8 @@ def read(self):
         self.channels = {'2lss_2b': self.ch2lss2b, '2lss_2b_p': self.ch2lss2b_p, '2lss_2b_m': self.ch2lss2b_m,'2lss_3b': self.ch2lss3b, '2lss_3b_p': self.ch2lss3b_p, '2lss_3b_m': self.ch2lss3b_m,'2lss_4b': self.ch2lss4b, '2lss_4b_p': self.ch2lss4b_p, '2lss_4b_m': self.ch2lss4b_m, '3l_1b': self.ch3l1b, '3l_1b_p': self.ch3l1b_p, '3l_1b_m': self.ch3l1b_m, '3l_p_offZ_1b': self.ch3l1b_p, '3l_m_offZ_1b': self.ch3l1b_m, '3l_p_offZ_2b': self.ch3l2b_p, '3l_m_offZ_2b': self.ch3l2b_m, '3l_2b': self.ch3l2b, '3l_2b_p': self.ch3l2b_p, '3l_2b_m': self.ch3l2b_m, '3l_sfz': self.ch3lsfz, '3l_sfz_1b': self.ch3lsfz1b, '3l_sfz_2b': self.ch3lsfz2b, '3l_onZ_1b': self.ch3lsfz1b, '3l_onZ_2b': self.ch3lsfz2b, '4l': self.ch4l}
         self.skip_process_channels = {**self.skip_process_channels, **{'data': [k for k in self.channels]}} # Skip all data!
         self.skip_process_channels = {**self.skip_process_channels, **{'flips': [k for k in self.channels if '2l' not in k]}} # Charge flips only in 2lss channels
-
+	self.skip_process_channels['data'].extend(['2lss', '2lss_m', '2lss_p'])
```
[the remark this answers is not in the document]
Comment (bryates) [reply analysis/topEFT/datacard_maker.py:147]: That's a good question. It probably will, so I'll look into it more.

**Comment by kmohrman** [comment]
@Dovahkinsl, should we close this PR?

**Pull request closed without merging**